dao/src/main/java/org/thingsboard/server/dao/sql/alarm/AlarmRepository.java

```diff
@@ -283,10 +273,8 @@ Page<AlarmInfoEntity> findCustomerAlarms(@Param("tenantId") UUID tenantId,
                     "WHERE a.tenantId = :tenantId AND a.customerId = :customerId " +
                     "AND (:startTime IS NULL OR a.createdTime >= :startTime) " +
                     "AND (:endTime IS NULL OR a.createdTime <= :endTime) " +
-                    "AND ((:#{#alarmTypes == null} = true) OR a.type IN (:alarmTypes)) " + //HHH-15968
-                    "AND ((:#{#alarmSeverities == null} = true) OR a.severity IN (:alarmSeverities)) " + //HHH-15968
-//                    "AND ((:alarmTypes) IS NULL OR a.type IN (:alarmTypes)) " +
-//                    "AND ((:alarmSeverities) IS NULL OR a.severity IN (:alarmSeverities)) " +
+                    "AND ((:alarmTypes) IS NULL OR a.type IN (:alarmTypes)) " +
+                    "AND ((:alarmSeverities) IS NULL OR a.severity IN (:alarmSeverities)) " +
```

Comment:
Why is this changed? Is HHH-15968 no longer an issue?

Reply:
yes it is fixed and new jpa version is being used in the project